sphinx_immaterial/content_tabs.py

```diff
@@ -0,0 +1,207 @@
+    }
+
+    def run(self) -> List[nodes.Node]:
+        """Run the directive."""
+        self.assert_has_content()
+        if not is_md_tab_type(self.state_machine.node, "md-tab-set"):
+            LOGGER.warning(
+                "The parent of a 'md-tab-item' should be a 'md-tab-set'",
+                location=(self.env.docname, self.lineno),
+            )
+        tab_item = nodes.container(
+            "", is_div=True, type="md-tab-item", classes=["tabbed-block"]
+        )
+
+        # add tab label
+        textnodes, _ = self.state.inline_text(self.arguments[0], self.lineno)
+        tab_label = nodes.rubric(
+            self.arguments[0],
+            *textnodes,
+            classes=["tabbed-label"],
+        )
+        self.add_name(tab_label)
+        tab_item += tab_label
+
+        # add tab content
+        tab_content = nodes.container(
+            "",
+            is_div=True,
+            type="",
+            classes=["tabbed-block"] + self.options.get("class", []),
+        )
+        self.state.nested_parse(self.content, self.content_offset, tab_content)
+        tab_item += tab_content
+
+        return [tab_item]
+
+
+class content_tab_input(nodes.Element, nodes.General):
+    pass
+
+
+class content_tab_label(nodes.TextElement, nodes.General):
+    pass
+
+
+def visit_tab_input(self, node):
+    attributes = {"ids": [node["id"]], "type": node["type"], "name": node["set_id"]}
+    if node["checked"]:
+        attributes["checked"] = "checked"
+    self.body.append(self.starttag(node, "input", **attributes))
+
+
+def depart_tab_input(self, node):
+    self.body.append("</input>")
+
+
+def visit_tab_label(self, node):
+    attributes = {"for": node["input_id"]}
+    self.body.append(self.starttag(node, "label", **attributes))
+
+
+def depart_tab_label(self, node):
+    self.body.append("</label>")
+
+
+def visit_tab_set(self: HTMLTranslator, node: content_tab_set):
+    # increment tab set counter
+    self.tab_set_count = getattr(self, "tab_set_count", 0) + 1
+
+    # configure tab set's div attributes
+    tab_set_identity = "__tabbed_" + str(self.tab_set_count)
+    attributes = {"data-tabs": str(self.tab_set_count) + node["data-tabs"]}
+    node["set_id"] = tab_set_identity
+    self.body.append(self.starttag(node, "div", **attributes))
+
+    # walkabout the children
+    tab_total = 0
+    children = []
+    tab_label_div = nodes.container("", is_div=True, classes=["tabbed-labels"])
+    tab_content_div = nodes.container("", is_div=True, classes=["tabbed-content"])
+    for tab_item in node.children:
+        try:
+            tab_label, tab_block = tab_item.children
+        except ValueError as exc:
+            raise ValueError(f"md-tab-item has no children:\n{repr(tab_item)}") from exc
+        tab_total += 1
+        tab_item_identity = tab_set_identity + f"_{tab_total}"
+
+        # create: <input checked="checked" id="id" type="radio">
+        input_node = content_tab_input(
```

Comment:
You could do this tag more simply as:

```
self.body.append(f'<input id="{self.attval(tab_item_identity)}" name="{self.attval(tab_set_identity)}" type="radio"{" checked" if tab_total == 1 else ""}>')
```

I think that more clearly indicates what is happening and avoids the need for the extra content_tab_input node.

Reply:
Again, this assumes that what the user entered is well formed. I think by doing it the current way, it is clearer, and string normalization (including exceptions raised about invalid characters) from docutils is used.

I understand that there are other methods that can be called to take advantage of docutils conveniences, but this just seems like the simplest and conventional.

Reply:
After giving this more thought, The conveniences of docutils in this instance shouldn't be needed. I applied this suggestion, but I still feel it isn't the most robust.